seatsio-ios/SeatingChartConfig.swift

```diff
@@ -1,83 +1,84 @@
-    var showSectionPricingOverlay: Bool?
-    var channels: Set<String>?
-
-    var onSelectionValid: Optional<() -> ()>
-    var onSelectionInvalid: Optional<([SelectionValidatorType]) -> ()>
-    var onObjectSelected: Optional<(SeatsioObject, TicketType?) -> ()>
-    var onObjectDeselected: Optional<(SeatsioObject, TicketType?) -> ()>
-    var onObjectClicked: Optional<(SeatsioObject) -> ()>
-    var onObjectStatusChanged: Optional<(SeatsioObject) -> ()>
-    var onSessionInitialized: Optional<(HoldToken) -> ()>
-    var onHoldTokenExpired: Optional<() -> ()>
-    var onBestAvailableSelected: Optional<([SeatsioObject], Bool) -> ()>
-    var onBestAvailableSelectionFailed: Optional<() -> ()>
-    var onHoldSucceeded: Optional<([SeatsioObject], [TicketType]?) -> ()>
-    var onHoldFailed: Optional<([SeatsioObject], [TicketType]?) -> ()>
-    var onReleaseHoldSucceeded: Optional<([SeatsioObject], [TicketType]?) -> ()>
-    var onReleaseHoldFailed: Optional<([SeatsioObject], [TicketType]?) -> ()>
-    var onSelectedObjectBooked: Optional<(SeatsioObject) -> ()>
-    var onChartRendered: Optional<(SeatingChart) -> ()>
-    var onChartRenderingFailed: Optional<() -> ()>
+public class SeatingChartConfig: SeatsioConfig, Encodable {
+
+    public private(set) var workspaceKey: String?
+    public private(set) var events: Set<String>?
+    public private(set) var chart: String?
+    public private(set) var pricing: [Pricing]?
+
+    public private(set) var priceFormatter: ((Float) -> String)?
+    public private(set) var numberOfPlacesToSelect: Int?
+    public private(set) var objectWithoutPricingSelectable: Bool?
+    public private(set) var selectedObjects: [SelectedObject]?
+    public private(set) var selectableObjects: [String]?
+    public private(set) var language: String?
+    public private(set) var messages: [String: String]?
+    public private(set) var priceLevelsTooltipMessage: String?
+    public private(set) var maxSelectedObjects: AnyEncodable?
+    public private(set) var availableCategories: [String]?
+    public private(set) var unavailableCategories: [String]?
+    public private(set) var filteredCategories: [String]?
+    public private(set) var categoryFilter: CategoryFilter?
+    public private(set) var selectBestAvailable: BestAvailable?
+    public private(set) var alwaysShowSectionContents: Bool?
+    public private(set) var showSectionContents: String?
+    public private(set) var showLegend: Bool?
+    public private(set) var showSeatLabels: Bool?
+    public private(set) var legend: Legend?
+    public private(set) var showMinimap: Bool?
+    public private(set) var showActiveSectionTooltip: Bool?
+    public private(set) var showViewFromYourSeat: Bool?
+    public private(set) var selectionValidators: [SelectionValidator]?
+    public private(set) var categories: [Category]?
+    public private(set) var objectCategories: [String: String]?
+    public private(set) var mode: EventMode?
```

Comment:
It would be best to have 2 sets of modes: 1 set for the renderer  (`SeatingChartConfig`), and 1 for the event manager (`EventManagerConfig`).

https://docs.seats.io/docs/renderer/config-mode/
https://docs.seats.io/docs/event-manager/modes

Reply:
Thanks @mroloux. I have created a different enum for SeatingChart.